src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocket.cs

```diff
@@ -441,6 +448,12 @@ private Task<bool> InternalReceiveAsync(ArraySegment<byte> buffer)
                 Abort();
             }
 
+            // If the task has completed, there's no need to force continuations to run async
+            if (_operation.TcsReceive.Task.Status == TaskStatus.RanToCompletion)
+            {
+                return Task.FromResult(_operation.TcsReceive.Task.GetAwaiter().GetResult());
+            }
```

Comment:
Can drop the `if` block then; all its doing is more work to return a completed task which `_operation.TcsReceive.Task` is anyway?